src/graph/rings.cc

```diff
@@ -6,17 +6,13 @@
 
 #include "core.h"
 
-#define MAXWIDTH 20
-#define PREFIXLEN 15
-#define STRLENGTH (PREFIXLEN+5*MAXWIDTH)
 void dumpLine(int* values, int nranks, const char* prefix) {
-  int prefixlen = strlen(prefix);
-  char line[STRLENGTH+1];
-  line[STRLENGTH] = '\0';
-  memset(line, ' ', STRLENGTH);
-  strncpy(line, prefix, PREFIXLEN);
-  for (int i=0; i<nranks && i<MAXWIDTH; i++) sprintf(line+prefixlen+4*i, " %3d", values[i]);
-  INFO(NCCL_INIT,"%s", line);
+  char line[128];
+  int n = snprintf(line, sizeof(line), "%s", prefix);
+  for (int i = 0; i < nranks && sizeof(line) > n; i++) {
+    n += snprintf(line + n, sizeof(line) - n, " %d", values[i]);
```

Comment:
Sorry I forgot the most important comment I wanted to make: we used to print with `%3d` so that numbers would be aligned between rings (at least with <1000 ranks) which was quite helpful for visual inspection, to check how rings were created. Was there any issue with `%3d`?